packages/trpc/server/routers/viewer/bookings/reportBooking.handler.ts

```diff
@@ -0,0 +1,233 @@
+        teams: {
+          some: {
+            team: {
+              members: {
+                some: {
+                  id: { in: membershipIdsWhereUserIsAdminOwner },
+                },
+              },
+            },
+          },
+        },
+      },
+      select: {
+        email: true,
+      },
+    });
+
+    const teamMemberEmailList = teamMemberEmails.map((user) => user.email);
+
+    // Check if any booking attendees are team members where user is admin/owner
+    const hasTeamMemberAttendee = booking.attendees.some((attendee) =>
+      teamMemberEmailList.includes(attendee.email)
+    );
+
+    // Check if booking owner is a team member where user is admin/owner
+    let isBookingOwnerTeamMember = false;
+    if (booking.userId) {
+      const bookingOwner = await prisma.user.findUnique({
+        where: { id: booking.userId },
+        select: { email: true },
+      });
+      isBookingOwnerTeamMember = !!(bookingOwner && teamMemberEmailList.includes(bookingOwner.email));
+    }
+
+    canAccessThroughTeamMembership = hasTeamMemberAttendee || isBookingOwnerTeamMember || isTeamAdminOrOwner;
+  }
+
+  if (!isBookingOwner && !isAttendee && !canAccessThroughTeamMembership) {
+    throw new TRPCError({ code: "FORBIDDEN", message: "You don't have access to this booking" });
+  }
+
+  // Check if user has already reported this booking or any in the recurring series
+  const existingReports = booking.reports;
+
+  if (booking.recurringEventId) {
+    // For recurring bookings, check if user has reported any booking in the series
+    const recurringReports = await prisma.bookingReport.findMany({
+      where: {
+        reportedById: user.id,
+        booking: {
+          recurringEventId: booking.recurringEventId,
+        },
+      },
+    });
+
+    if (recurringReports.length > 0) {
+      throw new TRPCError({
+        code: "BAD_REQUEST",
+        message: "You have already reported this recurring booking series",
+      });
+    }
+  } else if (existingReports.length > 0) {
+    throw new TRPCError({ code: "BAD_REQUEST", message: "You have already reported this booking" });
+  }
+
+  // For recurring bookings, report all remaining instances
+  let reportedBookingIds: number[] = [bookingId];
+
+  if (booking.recurringEventId) {
+    // Get all future bookings in the recurring series
+    const futureRecurringBookings = await prisma.booking.findMany({
+      where: {
+        recurringEventId: booking.recurringEventId,
+      },
+      select: {
+        id: true,
+      },
+    });
+
+    reportedBookingIds = futureRecurringBookings.map((b) => b.id);
+  }
+
+  // Create reports for all relevant bookings
+  await prisma.bookingReport.createMany({
+    data: reportedBookingIds.map((id) => ({
+      bookingId: id,
+      reportedById: user.id,
+      reason,
+      description,
+      cancelled: cancelBooking,
+    })),
+  });
+
+  const bookingReport = { id: reportedBookingIds[0] }; // For backward compatibility
```

Comment:
why "backward compatibility"? 

`bookingReport` is misleading as it's the bookingId not reportId. `ReportedBooking` would be better